.github/workflows/ci.yaml

```
@@ -545,10 +584,49 @@ jobs:
         delete-merged: true
       if: ${{ env.skip_artifact_upload != 'true' }}
 
+  pytest-run-parallel:
+    strategy:
+      matrix:
+        runner-image: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: ['3.13t', '3.14t-dev']
+    runs-on: ${{ matrix.runner-image }}
+    steps:
+    - name: clone repo
+      uses: actions/checkout@v4
+
+    - name: install python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: build and install
+      run: |
+        python -m pip install pytest setuptools pytest-run-parallel
+        python -m pip install .
+
+    - name: run tests under pytest-run-parallel
+      run: |
+        python -m pytest --parallel-threads=4
```

```suggestion
        python -m pytest --parallel-threads=4 src/c
```
I think testing on Windows is broken on upstream too, they only test src/c tests on Windows so we should do the same